packages/tools/devtools/devtools-browser-extension/src/devtools/TelemetryLogging.ts

```diff
@@ -0,0 +1,152 @@
+import { PostChannel, IChannelConfiguration, IXHROverride } from "@microsoft/1ds-post-js";
+import { ITelemetryBaseLogger, ITelemetryBaseEvent } from "@fluid-experimental/devtools-view";
+import { ITaggedTelemetryPropertyType } from "@fluidframework/core-interfaces";
+
+const fetchHttpXHROverride: IXHROverride = {
+	sendPOST: (payload, oncomplete, sync) => {
+		const telemetryRequestData =
+			typeof payload.data === "string"
+				? payload.data
+				: new TextDecoder().decode(payload.data);
+
+		const requestInit: RequestInit = {
+			body: telemetryRequestData,
+			method: "POST",
+			headers: payload.headers,
+			credentials: "include",
+		};
+		fetch(payload.urlString, requestInit)
+			.then((response) => {
+				const headerMap: { [key: string]: string } = {};
+				// response.headers is not a run-of-the-mill array, it satisfies a particular interface that
+				// only has forEach, not a general iterator.
+				// eslint-disable-next-line unicorn/no-array-for-each
+				response.headers.forEach((value: string, name: string) => {
+					headerMap[name] = value;
+				});
+
+				if (response.body) {
+					response
+						.text()
+						.then((text) => {
+							oncomplete(response.status, headerMap, text);
+						})
+						.catch((error) => {
+							// Something wrong with the response body? Play it safe by passing the response status; don't try to
+							// explicitly re-send the telemetry events by specifying status 0.
+							oncomplete(response.status, headerMap, "");
+						});
+				} else {
+					oncomplete(response.status, headerMap, "");
+				}
+			})
+			.catch((error) => {
+				console.error("Error issuing telemetry request:", error);
+				// Error sending the request. Set the status to 0 so that the events can be retried.
+				oncomplete(0, {});
+			});
+	},
+};
+/**
+ * Logger that sends logs to the OneDS collection endpoint.
+ */
+export class OneDSLogger implements ITelemetryBaseLogger {
+	private readonly appInsightsCore = new AppInsightsCore();
+	private readonly postChannel: PostChannel = new PostChannel();
+
+	/**
+	 * Controls whether this class initializes and uses any necessary underlying objects that send logs to a remote
+	 * endpoint. It's only set to true when an instrumentation key is provided, so no attempts are made to issue any
+	 * requests during local development or other scenarios where a key is not passed in.
+	 */
+	private readonly enabled: boolean = false;
+
+	public constructor() {
+		const channelConfig: IChannelConfiguration = {
+			alwaysUseXhrOverride: true,
+			httpXHROverride: fetchHttpXHROverride,
+		};
+
+		// Configure App insights core to send to collector
+		const coreConfig: IExtendedConfiguration = {
+			// A non-empty instrumentation key needs to provided for the logger to do anything.
+			// AB#5167 tracks how to inject one at build time to keep it out of source control.
+			instrumentationKey: "",
+			endpointUrl: "https://mobile.events.data.microsoft.com/OneCollector/1.0",
```

Comment:
Where do we get this endpoint url? Why is it "mobile"? :raised_eyebrow:

Reply:
`browser.events.data.microsoft.com` works and is even the default so I'm just removing this from here :)